client/header/activity-panel/panels/inbox/card.js

```diff
@@ -39,39 +48,210 @@ class InboxNoteCard extends Component {
+							'woocommerce-admin'
+						) }
+					</p>
+					<div className="woocommerce-inbox-dismiss-confirmation_buttons">
+						<Button
+							isDefault
+							onClick={ () => this.closeDismissModal() }
+						>
+							{ __( 'Cancel', 'woocommerce-admin' ) }
+						</Button>
+						{ this.getDismissConfirmationButton() }
+					</div>
+				</div>
+			</Modal>
+		);
+	}
+
 	render() {
 		const { lastRead, note } = this.props;
+		const { isDismissModalOpen } = this.state;
+		const {
+			actions: noteActions,
+			content,
+			date_created: dateCreated,
+			date_created_gmt: dateCreatedGmt,
+			id: noteId,
+			image,
+			layout,
+			title,
+		} = note;
 
 		const getButtonsFromActions = () => {
-			if ( ! note.actions ) {
+			if ( ! noteActions ) {
 				return [];
 			}
-			return note.actions.map( ( action ) => (
-				<NoteAction key={ note.id } noteId={ note.id } action={ action } />
+			return noteActions.map( ( action ) => (
+				<NoteAction
+					key={ noteId }
+					noteId={ noteId }
+					action={ action }
+				/>
 			) );
 		};
 
+		const unread =
+			! lastRead ||
+			! dateCreatedGmt ||
+			new Date( dateCreatedGmt + 'Z' ).getTime() > lastRead;
+		const actions = getButtonsFromActions( note );
+		const actionsList = Array.isArray( actions ) ? actions : [ actions ];
+		const date = dateCreated;
+		const hasImage = layout !== 'plain' && layout !== '';
+		const cardClassName = classnames( 'woocommerce-inbox-message', layout, {
+			'message-is-unread': unread,
+		} );
+
 		return (
 			<VisibilitySensor onChange={ this.onVisible }>
-				<ActivityCard
-					className={ classnames( 'woocommerce-inbox-activity-card', {
-						actioned: note.status !== 'unactioned',
-					} ) }
-					title={ note.title }
-					date={ note.date_created }
-					icon={ <Gridicon icon={ note.icon } size={ 48 } /> }
-					unread={
-						! lastRead ||
-						! note.date_created_gmt ||
-						new Date( note.date_created_gmt + 'Z' ).getTime() >
-							lastRead
-					}
-					actions={ getButtonsFromActions( note ) }
-				>
-					<span
-						dangerouslySetInnerHTML={ sanitizeHTML( note.content ) }
-					/>
-				</ActivityCard>
+				<section className={ cardClassName }>
+					{ hasImage && (
+						<div className="woocommerce-inbox-message__image">
+							<img src={ image } alt="" />
+						</div>
+					) }
+					<div className="woocommerce-inbox-message__wrapper">
+						<div className="woocommerce-inbox-message__content">
+							{ date && (
+								<span className="woocommerce-inbox-message__date">
+									{ moment.utc( date ).fromNow() }
+								</span>
+							) }
+							<H className="woocommerce-inbox-message__title">
+								{ title }
+							</H>
+							<Section className="woocommerce-inbox-message__text">
+								<span
+									dangerouslySetInnerHTML={ sanitizeHTML(
+										content
+									) }
+								/>
+							</Section>
+						</div>
+						<div className="woocommerce-inbox-message__actions">
+							{ actions && (
```

Comment:
Optional: might lessen the logic in this render and make it a little easier to follow if we change this area to `renderActions( note )`.  Then the logic above around converting to an array can be handled there.

Reply:
The change sounds good to me. I added it in the commit [cf76fe7](https://github.com/woocommerce/woocommerce-admin/pull/4379/commits/cf76fe71d4b2f5b782e13ab2b48b73b0ffaaf6ea) in the [PR 4379 ](https://github.com/woocommerce/woocommerce-admin/pull/4379)